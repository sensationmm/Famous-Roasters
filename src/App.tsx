import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
