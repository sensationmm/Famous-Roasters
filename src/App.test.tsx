import { render, waitFor } from '@testing-library/react'
import React from 'react'

import App from './App'

window.scrollTo = jest.fn()

describe('App', () => {
  it.skip('Renders correctly', async () => {
    const { container } = render(<App />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
