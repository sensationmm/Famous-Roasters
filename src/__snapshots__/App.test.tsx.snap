// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders correctly 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col"
  >
    <div
      class="bg-white"
    >
      <header
        class="relative bg-white"
      >
        <nav
          aria-label="Top"
          class="max-w-7xl mx-auto px-4 sm:px-6 xl:px-8"
        >
          <div
            class="h-16 flex items-center justify-between"
          >
            <div
              class="flex-1 flex items-center xl:hidden"
            >
              <button
                class="-ml-2 bg-white p-2 rounded-md text-gray-400"
                data-testid="button-open"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Open menu
                </span>
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 6h16M4 12h16M4 18h16"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="hidden xl:flex-1 xl:block xl:self-stretch"
            >
              <div
                class="h-full flex space-x-8"
              >
                <a
                  class="flex items-center text-gray-700 hover:text-gray-800"
                  href="/taste-finder"
                >
                  <span
                    class="uppercase text-sm leading-5 font-normal"
                  >
                    Tastefinder
                  </span>
                </a>
                <a
                  class="flex items-center text-gray-700 hover:text-gray-800"
                  href="/our-roasters"
                >
                  <span
                    class="uppercase text-sm leading-5 font-normal"
                  >
                    Unsere Röster
                  </span>
                </a>
              </div>
            </div>
            <a
              class="flex h-full items-center pl-2 pr-2"
              href="/"
            >
              <span
                class="sr-only"
              >
                Famous Roasters
              </span>
              <div
                class="xl:hidden font-syne text-3xl"
              >
                <span>
                  F
                </span>
                <span
                  class="font-extrabold"
                >
                  R
                </span>
              </div>
              <div
                class="hidden xl:block font-syne text-xl text-center"
              >
                <div
                  class="h-5"
                >
                  Famous
                </div>
                <div
                  class="font-extrabold"
                >
                  Roasters
                </div>
              </div>
            </a>
            <div
              class="flex-1 flow-root flex h-full items-center"
            >
              <div
                class="flex-1 flex h-full items-center justify-end xl:flex-none"
              >
                <div
                  class="hidden h-full xl:flex space-x-8 justify-end"
                >
                  <a
                    class="flex items-center text-gray-700 hover:text-gray-800"
                    href="/about"
                  >
                    <span
                      class="uppercase text-sm leading-5 font-normal"
                    >
                      Über uns
                    </span>
                  </a>
                  <a
                    class="flex items-center text-gray-700 hover:text-gray-800"
                    href="/contact"
                  >
                    <span
                      class="uppercase text-sm leading-5 font-normal"
                    >
                      Kontakt
                    </span>
                  </a>
                </div>
                <a
                  class="hidden xl:block p-2 text-gray-400 hover:text-gray-500 xl:ml-6"
                  href="/account"
                >
                  <span
                    class="sr-only"
                  >
                    Konto
                  </span>
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </a>
                <div
                  class="ml-4 flow-root xl:ml-6"
                >
                  <button
                    class="inline-flex items-center px-4 py-2 text-sm text-white bg-brand-black rounded-full"
                    data-testid="button-shop"
                    role="button"
                  >
                    Shop
                  </button>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <main
      class="flex-grow flex items-center justify-center bg-brand-grey-whisper"
    >
      <div>
        <div
          class="font-syne flex justify-center text-4xl md:text-5xl xl:text-6xl"
        >
          <h1>
            <span>
              Famous
            </span>
             
            <span
              class="font-bold"
            >
              Roasters
            </span>
          </h1>
        </div>
        <div
          class="text-center text-lg leading-7 font-normal"
        >
          Home
        </div>
      </div>
    </main>
    <footer
      class="bg-white"
    >
      <div
        class="max-w-7xl mx-auto py-6 px-4 sm:px-6 md:flex md:items-center md:justify-between lg:px-8"
      >
        <div
          class="md:order-1"
        >
          <p
            class="text-center text-base text-gray-400"
          >
            © 2022 Famous Roasters. Alle Rechte vorbehalten.
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
